src/shogun/base/SGObject.h

```diff
@@ -448,6 +448,10 @@ class CSGObject
 	void put(const std::string& name, T* value)
 	{
 		put(Tag<T*>(name), value);
+
+		std::string callback_fun = name + "_callback";
+		if (has(callback_fun))
```

Comment:
@lisitsyn @vigsterkr What are your thoughts on this callback stuff? I am not sure I like it.. One alternative is to add `CSGObject::init_seed` which is NOP if there is no seed parameter, but I am not sure I like that either